lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -219,7 +218,7 @@ protected function getVarcharTypeDeclarationSQLSnippet($length, $fixed)
     /**
      * {@inheritdoc}
      */
-    protected function getBinaryTypeDeclarationSQLSnippet($length, $fixed)
+    protected function getBinaryTypeDeclarationSQLSnippet(?int $length, bool $fixed) : string
```

Comment:
Wouldn't `0` be better default?

Reply:
Actually, it shouldn't be nullable. For a field with the non-variable length, there's no point in having any default. I have a branch with some rework in this area.